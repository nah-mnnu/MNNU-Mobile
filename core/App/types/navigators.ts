import { NavigatorScreenParams } from '@react-navigation/core'

import { DeclineType } from './decline'

export enum Screens {
  AttemptLockout = 'Temporarily Locked',
  Splash = 'Splash',
  Onboarding = 'Onboarding',
  Terms = 'Terms',
  CreatePIN = 'Create a PIN',
  EnterPIN = 'Enter PIN',
  Home = 'Home',
  Scan = 'Scan',
  Credentials = 'Credentials',
  CredentialDetails = 'Credential Details',
  Notifications = 'Notifications',
  CredentialOffer = 'Credential Offer',
  ProofRequest = 'Proof Request',
  ProofRequestAttributeDetails = 'Proof Request Attribute Details',
  Settings = 'Settings',
  Language = 'Language',
  Contacts = 'Contacts',
  ContactDetails = 'Contact Details',
  Chat = 'Chat',
  Connection = 'Connection',
  OnTheWay = 'On The Way',
  Declined = 'Declined',
  CommonDecline = 'Common Decline',
  UseBiometry = 'Use Biometry',
  RecreatePIN = 'Change PIN',
  Developer = 'Developer',
}

export enum Stacks {
  TabStack = 'Tab Stack',
  HomeStack = 'Home Stack',
  ConnectStack = 'Connect Stack',
  CredentialStack = 'Credentials Stack',
  SettingStack = 'Settings Stack',
  ContactStack = 'Contacts Stack',
  NotificationStack = 'Notifications Stack',
  ConnectionStack = 'Connection Stack',
}

export enum TabStacks {
  HomeStack = 'Tab Home Stack',
  ConnectStack = 'Tab Connect Stack',
  CredentialStack = 'Tab Credential Stack',
}

export type RootStackParams = {
  [Screens.Splash]: undefined
  [Stacks.TabStack]: NavigatorScreenParams<TabStackParams>
  [Stacks.ConnectStack]: NavigatorScreenParams<ConnectStackParams>
  [Stacks.SettingStack]: NavigatorScreenParams<SettingStackParams>
  [Stacks.ContactStack]: NavigatorScreenParams<ContactStackParams>
  [Stacks.NotificationStack]: NavigatorScreenParams<NotificationStackParams>
}

export type TabStackParams = {
  [TabStacks.HomeStack]: NavigatorScreenParams<HomeStackParams>
  [TabStacks.ConnectStack]: NavigatorScreenParams<ConnectStackParams>
  [TabStacks.CredentialStack]: NavigatorScreenParams<CredentialStackParams>
}

export type AuthenticateStackParams = {
  [Screens.Onboarding]: undefined
  [Screens.Terms]: undefined
  [Screens.AttemptLockout]: undefined
  [Screens.CreatePIN]: { setAuthenticated: (status: boolean) => void } | undefined
  [Screens.EnterPIN]: { setAuthenticated: (status: boolean) => void } | undefined
  [Screens.UseBiometry]: undefined
}

export type ContactStackParams = {
  [Screens.Contacts]: undefined
  [Screens.Chat]: { connectionId: string }
  [Screens.ContactDetails]: { connectionId: string }
}

export type CredentialStackParams = {
  [Screens.Credentials]: undefined
  [Screens.CredentialDetails]: { credentialId: string }
}

export type HomeStackParams = {
  [Screens.Home]: undefined
  [Screens.Notifications]: undefined
}

export type ConnectStackParams = {
  [Screens.Scan]: undefined
}

export type SettingStackParams = {
  [Screens.Settings]: undefined
  [Screens.Language]: undefined
  [Screens.UseBiometry]: undefined
  [Screens.CreatePIN]: undefined
  [Screens.RecreatePIN]: undefined
  [Screens.Terms]: undefined
  [Screens.Onboarding]: undefined
  [Screens.Developer]: undefined
}

export type NotificationStackParams = {
  [Screens.CredentialDetails]: { credentialId: string }
  [Screens.CredentialOffer]: { credentialId: string }
  [Screens.ProofRequest]: { proofId: string }
  [Screens.ProofRequestAttributeDetails]: {
    proofId: string
    attributeName: string | null
  }
  [Screens.CommonDecline]: {
    declineType: DeclineType
    itemId: string
  }
}

export type DeliveryStackParams = {
  [Screens.Connection]: { connectionId?: string; threadId?: string }
  [Screens.CredentialOffer]: { credentialId: string }
  [Screens.ProofRequest]: { proofId: string }
  [Screens.ProofRequestAttributeDetails]: {
    proofId: string
    attributeName: string | null
  }
  [Screens.CommonDecline]: {
    declineType: DeclineType
    itemId: string
  }
  [Screens.OnTheWay]: { credentialId: string }
  [Screens.Declined]: { credentialId: string }
}
