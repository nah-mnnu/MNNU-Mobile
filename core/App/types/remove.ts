export enum RemoveType {
  Credential = 'Credential',
  Contact = 'Contact',
}
