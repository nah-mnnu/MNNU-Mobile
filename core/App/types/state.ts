import { BifoldError } from './error'

export interface Onboarding {
  didCompleteTutorial: boolean
  didAgreeToTerms: boolean
  didCreatePIN: boolean
  didConsiderBiometry: boolean
}

export interface Preferences {
  useBiometry: boolean
  developerModeEnabled: boolean
}

export interface Lockout {
  displayNotification: boolean
}

export interface LoginAttempt {
  lockoutDate?: number
  servedPenalty: boolean
  loginAttempts: number
}

export interface Privacy {
  didShowCameraDisclosure: boolean
}

export interface Authentication {
  didAuthenticate: boolean
}

export interface State {
  onboarding: Onboarding
  authentication: Authentication
  privacy: Privacy
  lockout: Lockout
  loginAttempt: LoginAttempt
  preferences: Preferences
  error: BifoldError | null
  loading: boolean
}
