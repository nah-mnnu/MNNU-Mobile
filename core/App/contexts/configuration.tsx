import { IndyPoolConfig } from '@aries-framework/core'
import { createContext, useContext } from 'react'

import { RecordProps } from '../components/record/Record'
import OnboardingPages from '../screens/OnboardingPages'
import { ScanProps } from '../screens/Scan'
import { OCABundleResolver } from '../types/oca'
import { PINSecurityParams } from '../types/security'
import { SettingSection } from '../types/settings'

export interface ConfigurationContext {
  pages: typeof OnboardingPages
  splash: React.FC
  terms: React.FC
  homeContentView: React.FC
  developer: React.FC
  OCABundle: OCABundleResolver
  scan: React.FC<ScanProps>
  useBiometry: React.FC
  record: React.FC<RecordProps>
  PINSecurity: PINSecurityParams
  indyLedgers: IndyPoolConfig[]
  settings: SettingSection[]
}

export const ConfigurationContext = createContext<ConfigurationContext>(null as unknown as ConfigurationContext)

export const ConfigurationProvider = ConfigurationContext.Provider

export const useConfiguration = () => useContext(ConfigurationContext)
