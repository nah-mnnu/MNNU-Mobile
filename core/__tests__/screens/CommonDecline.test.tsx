import {
  CredentialExchangeRecord as CredentialRecord,
  CredentialState,
  ProofRecord,
  ProofState,
} from '@aries-framework/core'
import { useCredentialById, useProofById, useAgent } from '@aries-framework/react-hooks'
import { useNavigation } from '@react-navigation/core'
import { render, fireEvent, waitFor } from '@testing-library/react-native'
import fs from 'fs'
import path from 'path'
import React from 'react'
import { act } from 'react-test-renderer'

import { ConfigurationContext } from '../../App/contexts/configuration'
import CommonDecline from '../../App/screens/CommonDecline'
import { DeclineType } from '../../App/types/decline'
import { testIdWithKey } from '../../App/utils/testable'
import configurationContext from '../contexts/configuration'

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper')
jest.mock('@react-navigation/core', () => {
  return require('../../__mocks__/custom/@react-navigation/core')
})
jest.mock('@react-navigation/native', () => {
  return require('../../__mocks__/custom/@react-navigation/native')
})
// eslint-disable-next-line @typescript-eslint/no-empty-function
jest.mock('react-native-localize', () => {})

const credentialPath = path.join(__dirname, '../fixtures/degree-credential.json')
const credential = JSON.parse(fs.readFileSync(credentialPath, 'utf8'))

const proofPath = path.join(__dirname, '../fixtures/degree-proof.json')
const proof = JSON.parse(fs.readFileSync(proofPath, 'utf8'))

describe('common decline screen', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })

  test('decline proof renders correctly', () => {
    // @ts-ignore
    useProofById.mockReturnValueOnce(new ProofRecord(proof))

    const props = { params: { declineType: DeclineType.ProofRequest, itemId: 'b04060c8-902a-4f62-9390-7b8bb2204f13' } }
    const tree = render(<CommonDecline route={props as any} navigation={useNavigation()} />)
    const confirmDeclineButton = tree.getByTestId(testIdWithKey('ConfirmDeclineButton'))
    const abortDeclineButton = tree.getByTestId(testIdWithKey('AbortDeclineButton'))

    expect(tree).toMatchSnapshot()
    expect(confirmDeclineButton).not.toBeNull()
    expect(abortDeclineButton).not.toBeNull()
  })

  test('decline proof triggers AFJ', () => {
    // @ts-ignore
    useProofById.mockReturnValueOnce(new ProofRecord(proof))

    const props = {
      params: { declineType: DeclineType.ProofRequest, itemId: 'b04060c8-902a-4f62-9390-7b8bb2204f13' },
    }
    const { getByTestId } = render(<CommonDecline route={props as any} navigation={useNavigation()} />)
    const confirmDeclineButton = getByTestId(testIdWithKey('ConfirmDeclineButton'))
    const { agent } = useAgent()

    fireEvent(confirmDeclineButton, 'press')

    expect(agent?.proofs.declineRequest).toBeCalledWith('b04060c8-902a-4f62-9390-7b8bb2204f13')
  })

  test('did decline proof renders correctly', async () => {
    const rec = new ProofRecord(proof)
    rec.state = ProofState.Declined

    // @ts-ignore
    useProofById.mockReturnValueOnce(rec)

    const props = { params: { declineType: DeclineType.ProofRequest, itemId: 'b04060c8-902a-4f62-9390-7b8bb2204f13' } }
    const tree = render(<CommonDecline route={props as any} navigation={useNavigation()} />)
    const doneButton = tree.getByTestId(testIdWithKey('Done'))
    const didDeclineText = tree.getByTestId(testIdWithKey('RequestOrOfferDeclined'))

    expect(tree).toMatchSnapshot()
    expect(doneButton).not.toBeNull()
    expect(didDeclineText).not.toBeNull()
  })

  test('navigate home after decline proof', async () => {
    const rec = new ProofRecord(proof)
    rec.state = ProofState.Declined

    // @ts-ignore
    useProofById.mockReturnValueOnce(rec)

    const props = { params: { declineType: DeclineType.ProofRequest, itemId: 'b04060c8-902a-4f62-9390-7b8bb2204f13' } }
    const { getByTestId } = render(<CommonDecline route={props as any} navigation={useNavigation()} />)
    const doneButton = getByTestId(testIdWithKey('Done'))

    fireEvent(doneButton, 'press')

    const navigation = useNavigation()

    expect(navigation.navigate).toBeCalledWith('Tab Home Stack', { screen: 'Home' })
  })

  test('decline offer renders correctly', async () => {
    const rec = new CredentialRecord(credential)
    rec.credentials.push({
      credentialRecordType: 'indy',
      credentialRecordId: '',
    })
    // TODO:(jl) Make a fn to revive JSON dates properly and pass to `parse`
    rec.createdAt = new Date(rec.createdAt)

    // @ts-ignore
    useCredentialById.mockReturnValueOnce(rec)

    const props = {
      params: { declineType: DeclineType.CredentialOffer, itemId: '0683de72-2d24-4c76-a471-424c832e4b93' },
    }
    const tree = render(
      <ConfigurationContext.Provider value={configurationContext}>
        <CommonDecline route={props as any} navigation={useNavigation()} />
      </ConfigurationContext.Provider>
    )

    await act(async () => {
      // wait for appearance inside an assertion
      await waitFor(
        () => {
          expect(tree.getByTestId(testIdWithKey('ShowCredentialDetails'))).toBeDefined()
        },
        { timeout: 50000 }
      )
      const confirmDeclineButton = tree.getByTestId(testIdWithKey('ConfirmDeclineButton'))
      const abortDeclineButton = tree.getByTestId(testIdWithKey('AbortDeclineButton'))

      expect(confirmDeclineButton).not.toBeNull()
      expect(abortDeclineButton).not.toBeNull()
      expect(tree).toMatchSnapshot()
    })
  })

  test('decline offer triggers AFJ', async () => {
    const rec = new CredentialRecord(credential)
    // TODO:(jl) Make a fn to revive JSON dates properly and pass to `parse`
    rec.createdAt = new Date(rec.createdAt)

    // @ts-ignore
    useCredentialById.mockReturnValueOnce(rec)

    const props = {
      params: { declineType: DeclineType.CredentialOffer, itemId: '0683de72-2d24-4c76-a471-424c832e4b93' },
    }
    const { getByTestId } = render(
      <ConfigurationContext.Provider value={configurationContext}>
        <CommonDecline route={props as any} navigation={useNavigation()} />
      </ConfigurationContext.Provider>
    )

    await act(async () => {
      const confirmDeclineButton = getByTestId(testIdWithKey('ConfirmDeclineButton'))
      const { agent } = useAgent()

      fireEvent(confirmDeclineButton, 'press')

      expect(agent?.credentials.declineOffer).toBeCalledWith('0683de72-2d24-4c76-a471-424c832e4b93')
    })
  })

  test('did decline offer renders correctly', () => {
    const rec = new CredentialRecord(credential)
    rec.state = CredentialState.Declined
    // TODO:(jl) Make a fn to revive JSON dates properly and pass to `parse`
    rec.createdAt = new Date(rec.createdAt)

    // @ts-ignore
    useCredentialById.mockReturnValueOnce(rec)

    // @ts-ignore
    useCredentialById.mockReturnValueOnce(rec)

    const props = {
      params: { declineType: DeclineType.CredentialOffer, itemId: '0683de72-2d24-4c76-a471-424c832e4b93' },
    }
    const tree = render(
      <ConfigurationContext.Provider value={configurationContext}>
        <CommonDecline route={props as any} navigation={useNavigation()} />
      </ConfigurationContext.Provider>
    )
    const doneButton = tree.getByTestId(testIdWithKey('Done'))
    const didDeclineText = tree.getByTestId(testIdWithKey('RequestOrOfferDeclined'))

    expect(tree).toMatchSnapshot()
    expect(doneButton).not.toBeNull()
    expect(didDeclineText).not.toBeNull()
  })

  test('navigate home after decline offer', () => {
    const rec = new CredentialRecord(credential)
    rec.state = CredentialState.Declined
    // TODO:(jl) Make a fn to revive JSON dates properly and pass to `parse`
    rec.createdAt = new Date(rec.createdAt)

    // @ts-ignore
    useCredentialById.mockReturnValueOnce(rec)

    const props = {
      params: { declineType: DeclineType.CredentialOffer, itemId: '0683de72-2d24-4c76-a471-424c832e4b93' },
    }
    const { getByTestId } = render(
      <ConfigurationContext.Provider value={configurationContext}>
        <CommonDecline route={props as any} navigation={useNavigation()} />
      </ConfigurationContext.Provider>
    )
    const doneButton = getByTestId(testIdWithKey('Done'))

    fireEvent(doneButton, 'press')

    const navigation = useNavigation()

    expect(navigation.navigate).toBeCalledWith('Tab Home Stack', { screen: 'Home' })
  })
})
