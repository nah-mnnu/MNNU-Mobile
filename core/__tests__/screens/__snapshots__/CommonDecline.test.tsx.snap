// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common decline screen decline offer renders correctly 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "left",
      "right",
      "bottom",
    ]
  }
>
  <RCTScrollView
    style={
      Array [
        Object {
          "backgroundColor": "#000000",
          "height": "100%",
          "padding": 20,
        },
      ]
    }
  >
    <View>
      <View
        style={
          Object {
            "backgroundColor": "#313132",
            "borderColor": "#FCBA19",
            "borderRadius": 5,
            "borderWidth": 1,
            "minWidth": 700,
            "padding": 10,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "paddingHorizontal": 5,
              "paddingTop": 5,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "marginRight": 10,
                },
              ]
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#FCBA19",
                    "fontSize": 30,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              î€‚
            </Text>
          </View>
          <Text
            style={
              Object {
                "alignSelf": "center",
                "color": "#FFFFFF",
                "flexShrink": 1,
                "fontSize": 18,
                "fontWeight": "bold",
              }
            }
            testID="com.ariesbifold:id/HeaderText"
          >
            CredentialOffer.ConfirmDeclinedTitle
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "column",
              "marginLeft": 40,
              "paddingBottom": 5,
              "paddingHorizontal": 5,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "flexShrink": 1,
                "fontSize": 18,
                "fontWeight": "normal",
                "marginVertical": 16,
              }
            }
            testID="com.ariesbifold:id/BodyText"
          >
            CredentialOffer.ConfirmDeclinedMessage
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "marginTop": 20,
          }
        }
      >
        <RNGestureHandlerButton
          collapsable={false}
          onGestureEvent={[Function]}
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          onHandlerStateChange={[Function]}
          rippleColor={0}
          testID="com.ariesbifold:id/ShowCredentialDetails"
        >
          <View
            accessible={true}
            style={
              Object {
                "backgroundColor": "#15C477",
                "borderRadius": 15,
                "height": 375,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "flexGrow": 1,
                }
              }
              testID="com.ariesbifold:id/CredentialCard"
            >
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "rgba(0,0,0,0)",
                      "borderTopLeftRadius": 15,
                      "borderTopRightRadius": 15,
                      "flexDirection": "column",
                      "height": 101.75,
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "rgba(0,0,0,0)",
                        "flexDirection": "row",
                        "height": 93.75,
                        "marginBottom": 8,
                        "marginLeft": 8,
                        "marginRight": 8,
                        "marginTop": 8,
                      },
                    ]
                  }
                  testID="com.ariesbifold:id/CredentialCardHeader"
                >
                  <Text
                    ellipsizeMode="tail"
                    maxFontSizeMultiplier={1}
                    numberOfLines={1}
                    style={
                      Array [
                        Object {
                          "color": "#FFFFFF",
                          "fontSize": 14,
                          "fontWeight": "bold",
                        },
                        Object {
                          "color": "#313132",
                          "flex": 4,
                          "paddingHorizontal": 5,
                          "textAlignVertical": "center",
                        },
                      ]
                    }
                    testID="com.ariesbifold:id/CredentialIssuer"
                  >
                    d7e7f641-e0cc-46bc-908a-f351e2486b03
                  </Text>
                  <Text
                    ellipsizeMode="tail"
                    maxFontSizeMultiplier={1}
                    numberOfLines={1}
                    style={
                      Array [
                        Object {
                          "color": "#FFFFFF",
                          "fontSize": 14,
                          "fontWeight": "bold",
                        },
                        Object {
                          "color": "#313132",
                          "flex": 4,
                          "paddingHorizontal": 5,
                          "textAlign": "right",
                          "textAlignVertical": "center",
                        },
                      ]
                    }
                    testID="com.ariesbifold:id/CredentialName"
                  >
                    Credential
                  </Text>
                </View>
              </View>
              <View
                style={
                  Object {
                    "flexGrow": 1,
                  }
                }
                testID="com.ariesbifold:id/CredentialCardBody"
              />
              <View
                style={
                  Object {
                    "backgroundColor": "rgba(0,0,0,0)",
                    "borderBottomLeftRadius": 15,
                    "borderBottomRightRadius": 15,
                    "flexDirection": "row",
                    "paddingHorizontal": 10,
                    "paddingVertical": 10,
                  }
                }
                testID="com.ariesbifold:id/CredentialCardFooter"
              >
                <Text
                  maxFontSizeMultiplier={1}
                  style={
                    Array [
                      Object {
                        "color": "#FFFFFF",
                        "fontSize": 14,
                        "fontWeight": "normal",
                      },
                      Object {
                        "color": "#313132",
                      },
                    ]
                  }
                  testID="com.ariesbifold:id/CredentialIssued"
                >
                  CredentialDetails.Issued
                  : 
                  Feb 22, 2022
                </Text>
              </View>
            </View>
          </View>
        </RNGestureHandlerButton>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "margin": 20,
        "marginTop": "auto",
      }
    }
  >
    <View
      style={
        Object {
          "paddingTop": 10,
        }
      }
    >
      <View
        accessibilityLabel="CredentialOffer.ConfirmDecline"
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#42803E",
            "borderRadius": 4,
            "opacity": 1,
            "padding": 16,
          }
        }
        testID="com.ariesbifold:id/ConfirmDeclineButton"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
                false,
              ]
            }
          >
            CredentialOffer.ConfirmDecline
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "paddingTop": 10,
          },
        ]
      }
    >
      <View
        accessibilityLabel="CredentialOffer.AbortDecline"
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderColor": "#42803E",
            "borderRadius": 4,
            "borderWidth": 2,
            "opacity": 1,
            "padding": 16,
          }
        }
        testID="com.ariesbifold:id/AbortDeclineButton"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#42803E",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
                false,
              ]
            }
          >
            CredentialOffer.AbortDecline
          </Text>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;

exports[`common decline screen decline proof renders correctly 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "left",
      "right",
      "bottom",
    ]
  }
>
  <RCTScrollView
    style={
      Array [
        Object {
          "backgroundColor": "#000000",
          "height": "100%",
          "padding": 20,
        },
      ]
    }
  >
    <View>
      <View
        style={
          Object {
            "backgroundColor": "#313132",
            "borderColor": "#FCBA19",
            "borderRadius": 5,
            "borderWidth": 1,
            "minWidth": 700,
            "padding": 10,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "paddingHorizontal": 5,
              "paddingTop": 5,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "marginRight": 10,
                },
              ]
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#FCBA19",
                    "fontSize": 30,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              î€‚
            </Text>
          </View>
          <Text
            style={
              Object {
                "alignSelf": "center",
                "color": "#FFFFFF",
                "flexShrink": 1,
                "fontSize": 18,
                "fontWeight": "bold",
              }
            }
            testID="com.ariesbifold:id/HeaderText"
          >
            ProofRequest.ConfirmDeclinedTitle
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "column",
              "marginLeft": 40,
              "paddingBottom": 5,
              "paddingHorizontal": 5,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "flexShrink": 1,
                "fontSize": 18,
                "fontWeight": "normal",
                "marginVertical": 16,
              }
            }
            testID="com.ariesbifold:id/BodyText"
          >
            ProofRequest.ConfirmDeclinedMessage
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "margin": 20,
        "marginTop": "auto",
      }
    }
  >
    <View
      style={
        Object {
          "paddingTop": 10,
        }
      }
    >
      <View
        accessibilityLabel="ProofRequest.ConfirmDecline"
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#42803E",
            "borderRadius": 4,
            "opacity": 1,
            "padding": 16,
          }
        }
        testID="com.ariesbifold:id/ConfirmDeclineButton"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
                false,
              ]
            }
          >
            ProofRequest.ConfirmDecline
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "paddingTop": 10,
          },
        ]
      }
    >
      <View
        accessibilityLabel="ProofRequest.AbortDecline"
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderColor": "#42803E",
            "borderRadius": 4,
            "borderWidth": 2,
            "opacity": 1,
            "padding": 16,
          }
        }
        testID="com.ariesbifold:id/AbortDeclineButton"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#42803E",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
                false,
              ]
            }
          >
            ProofRequest.AbortDecline
          </Text>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;

exports[`common decline screen did decline offer renders correctly 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "left",
      "right",
      "bottom",
      "top",
    ]
  }
>
  <RCTScrollView
    style={
      Array [
        Object {
          "backgroundColor": "#000000",
          "height": "100%",
          "padding": 20,
        },
      ]
    }
  >
    <View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "marginTop": 25,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 26,
                "fontWeight": "bold",
              },
              Object {
                "fontWeight": "normal",
                "marginTop": 30,
                "textAlign": "center",
              },
            ]
          }
          testID="com.ariesbifold:id/RequestOrOfferDeclined"
        >
          CredentialOffer.CredentialDeclined
        </Text>
        <
          fill="#FFFFFF"
          height={250}
          style={
            Array [
              Object {
                "marginVertical": 66,
              },
            ]
          }
          width={250}
        />
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "margin": 20,
        "marginTop": "auto",
      }
    }
  >
    <View
      accessibilityLabel="Global.Done"
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#42803E",
          "borderRadius": 4,
          "opacity": 1,
          "padding": 16,
        }
      }
      testID="com.ariesbifold:id/Done"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 18,
                "fontWeight": "bold",
                "textAlign": "center",
              },
              false,
            ]
          }
        >
          Global.Done
        </Text>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;

exports[`common decline screen did decline proof renders correctly 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "left",
      "right",
      "bottom",
      "top",
    ]
  }
>
  <RCTScrollView
    style={
      Array [
        Object {
          "backgroundColor": "#000000",
          "height": "100%",
          "padding": 20,
        },
      ]
    }
  >
    <View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "marginTop": 25,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 26,
                "fontWeight": "bold",
              },
              Object {
                "fontWeight": "normal",
                "marginTop": 30,
                "textAlign": "center",
              },
            ]
          }
          testID="com.ariesbifold:id/RequestOrOfferDeclined"
        >
          ProofRequest.ProofRequestDeclined
        </Text>
        <
          fill="#FFFFFF"
          height={250}
          style={
            Array [
              Object {
                "marginVertical": 66,
              },
            ]
          }
          width={250}
        />
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "margin": 20,
        "marginTop": "auto",
      }
    }
  >
    <View
      accessibilityLabel="Global.Done"
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#42803E",
          "borderRadius": 4,
          "opacity": 1,
          "padding": 16,
        }
      }
      testID="com.ariesbifold:id/Done"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 18,
                "fontWeight": "bold",
                "textAlign": "center",
              },
              false,
            ]
          }
        >
          Global.Done
        </Text>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;
